src/actions/firebase/firebase.ts

```diff
@@ -0,0 +1,167 @@
+	import * as Hub from "../../hub"
+	import * as admin from 'firebase-admin';
+	import * as app from 'firebase-admin/app';
+
+	export class FirebaseAction extends Hub.Action {
+
+	  name = "firebase"
+	  label = "Firebase"
+	  iconName = "airtable/airtable.png"
+	  description = "Use firebase to send push notifications to mobile."
+	  storageBucket = 'gs://looker-9d4d2.appspot.com'
+	  contentType = "image/jpeg"
+	  notificationOptions = {
+      priority: "high",
+      timeToLive: 60 * 60 * 24,
+      contentAvailable: true,
+      mutableContent: true
+	  }
+	  static firebaseAdmin: app.App
+	  params = []
+	  supportedActionTypes = [Hub.ActionType.Query]
+	  supportedFormats = [Hub.ActionFormat.JsonDetail]
+	  supportedFormattings = [Hub.ActionFormatting.Unformatted]
+	  supportedVisualizationFormattings = [Hub.ActionVisualizationFormatting.Noapply]
+
+    static setFirebaseClient() {
+      if (!FirebaseAction.firebaseAdmin) {
+        FirebaseAction.firebaseAdmin = admin.initializeApp({
+        								credential: admin.credential.cert({projectId: process.env.FIREBASE_PROJECT_ID,
+        																   clientEmail: process.env.FIREBASE_CLIENT_EMAIL,
+        																   privateKey: process.env.FIREBASE_PRIVATE_KEY?.replace(/\\n/g, '\n')
+        																   }),
+        								databaseURL: process.env.FIREBASE_DATABASE,
+        							});
+      }
+    }
+
+	  async execute(request: Hub.ActionRequest) {
+      let response = new Hub.ActionResponse({success: true})
+      try {
+        let data: any = {}
+        if (request.formParams.data) {
+          data = request.formParams.data
+          if (!data.alertId) {
+            throw "Need Valid AlertId."
```

Comment:
just checking if this should throw or this should send a 404 with this as message ?
not sure if execute handles the exception to 404 part.